cli/azd/pkg/commands/pipeline/pipeline.go

```diff
@@ -84,75 +96,127 @@ func folderExists(folderPath string) bool {
 	return false
 }
 
+const (
+	gitHubLabel     string = "github"
+	githubFolder    string = ".github"
+	azdoLabel       string = "azdo"
+	azdoFolder      string = ".azdo"
+	envPersistedKey string = "AZD_PIPELINE_PROVIDER"
+)
+
 // DetectProviders get azd context from the context and pulls the project directory from it.
 // Depending on the project directory, returns pipeline scm and ci providers based on:
-// - if .github folder is found and .azdo folder is missing: GitHub scm and ci as provider
-// - if .azdo folder is found and .github folder is missing: Azdo scm and ci as provider
-// - both .github and .azdo folders found: prompt user to choose provider for scm and ci
-// - none of the folders found: return error
-// - no azd context in the ctx: return error
-func DetectProviders(ctx context.Context, console input.Console) (ScmProvider, CiProvider, error) {
+//   - if .github folder is found and .azdo folder is missing: GitHub scm and ci as provider
+//   - if .azdo folder is found and .github folder is missing: Azdo scm and ci as provider
+//   - both .github and .azdo folders found: GitHub scm and ci as provider
+//   - overrideProvider set to github (regardless of folders): GitHub scm and ci as provider
+//   - overrideProvider set to azdo (regardless of folders): Azdo scm and ci as provider
+//   - none of the folders found: return error
+//   - no azd context in the ctx: return error
+//   - overrideProvider set to neither github or azdo: return error
+//   - Note: The provider is persisted in the environment so the next time the function is run
+//     the same provider is used directly, unless the overrideProvider is used to change
+//     the last used configuration
+func DetectProviders(
+	ctx context.Context,
+	env *environment.Environment,
+	overrideProvider string) (ScmProvider, CiProvider, error) {
+	// This should be a pre-condition. azd context should be injected with the context
 	azdContext, err := azdcontext.GetAzdContext(ctx)
 	if err != nil {
 		return nil, nil, err
 	}
-
 	projectDir := azdContext.ProjectDirectory()
 
-	hasGitHubFolder := folderExists(path.Join(projectDir, ".github"))
-	hasAzDevOpsFolder := folderExists(path.Join(projectDir, ".azdo"))
+	// get the override value
+	overrideWith := strings.ToLower(overrideProvider)
 
+	// detecting pipeline folder configuration
+	hasGitHubFolder := folderExists(path.Join(projectDir, githubFolder))
+	hasAzDevOpsFolder := folderExists(path.Join(projectDir, azdoFolder))
+
+	// Error missing config for any provider
 	if !hasGitHubFolder && !hasAzDevOpsFolder {
-		return nil, nil, fmt.Errorf("no CI/CD provider configuration found. Expecting either .github and/or .azdo folder in the project root directory.")
+		return nil, nil, fmt.Errorf(
+			"no CI/CD provider configuration found. Expecting either %s and/or %s folder in the project root directory.",
+			gitHubLabel,
+			azdoLabel)
```

Comment:
With the addition of the override flag, do you think `azd` should be creating the directory instead of erroring out? Or are there additional configuration we depend on?

Reply:
calling `azd pipeline config` requires/must find either `.github` or `.azdo` folder at the root of the project directory.
There's no support from `azd` to create or pull these folders from some repository.

`yaml` file can be used to set the default provider to use, but it doesn't set the folder path.